src/QA/SoftMocks.php

```diff
@@ -949,16 +949,60 @@ public static function callFunction($namespace, $func, $params)
         return call_user_func_array($func, $params);
     }
 
-    public static function callNew($class, $args)
+    public static function callNew($class, $args, $scope)
     {
         if (isset(self::$new_mocks[$class])) {
             return call_user_func_array(self::$new_mocks[$class], $args);
         }
 
         $Rc = new \ReflectionClass($class);
-        return $Rc->newInstanceArgs($args);
+        $Constructor = $Rc->getConstructor();
+
+        if ($Constructor && self::isConstructorCanBeForced($Constructor, $scope)) {
+            $instance = $Rc->newInstanceWithoutConstructor();
+            $Constructor->setAccessible(true);
+            $Constructor->invokeArgs($instance, $args);
+            $Constructor->setAccessible(false);
```

Comment:
There is no need to set accesible back to false because setAccessible only affects ReflectionMethod instance, not the method itself